crates/ruff/src/rules/pep8_naming/rules/invalid_first_argument_name_for_class_method.rs

```diff
@@ -76,7 +76,7 @@ pub fn invalid_first_argument_name_for_class_method(
         return None;
     }
     if let Some(arg) = args.posonlyargs.first().or_else(|| args.args.first()) {
-        if arg.node.arg != "cls" {
+        if arg.node.arg.as_str() != "cls" {
```

Comment:
Would it make sense to also implement `PartialEq<str> for String` and `PartialEq<String> for &str` and (the same for `std::String`)?